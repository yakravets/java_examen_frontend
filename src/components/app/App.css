.site-page-header {
    border: 1px solid rgb(235, 237, 240);
}

.content {
    padding: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    top: 7%;
    bottom: 0;
    width: 100%;
}

.space_main{
    display: flex;
    justify-content: center;
    padding-bottom: 20px;
}