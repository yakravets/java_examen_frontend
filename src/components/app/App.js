import React, { Suspense } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import 'antd/dist/antd.css';
import './App.css';

import { Menu, Space } from 'antd';
import { AiOutlineInfoCircle } from "react-icons/ai";
import { GiCommercialAirplane } from "react-icons/gi";
import { MdAirlineSeatReclineExtra } from "react-icons/md";
import { FaRegListAlt } from "react-icons/fa";
import { GrMenu } from 'react-icons/gr';
import Loader from './components/Loader';
import { UserOutlined } from '@ant-design/icons';
import Login from './components/Login';

const Home = React.lazy(() => import('./components/Home'));
const About = React.lazy(() => import('./components/About'));
const ListTrips = React.lazy(() => import('./components/ListTrips'));
const ListAircraft = React.lazy(() => import('./components/ListAircraft'));
const ListAirports = React.lazy(() => import('./components/ListAirports'));

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {current: 'home'};
  }  

  handleClick = e => {
    this.setState({current: e.key});
  };

  render() {
    return (
      <Router>
        <Menu onClick={this.handleClick} selectedKeys={[this.state.current]} mode="horizontal">
          <Menu.Item key="home" icon={<GrMenu />}>
            <Link to="/">Home</Link>
          </Menu.Item>
          <Menu.Item key="trips" icon={<FaRegListAlt />}>
            <Link to="/trips">Trips</Link>
          </Menu.Item>
          <Menu.Item key="airports" icon={<MdAirlineSeatReclineExtra />}>
            <Link to="/airports">Airports</Link>
          </Menu.Item>
          <Menu.Item key="aircraft" icon={<GiCommercialAirplane />}>
            <Link to="/aircraft">Aircraft</Link>
          </Menu.Item>
          <Menu.Item key="about" icon={<AiOutlineInfoCircle/>}>
            <Link to="/about">About</Link>
          </Menu.Item> 
          <Menu.Item key="account" icon={<UserOutlined/>} className='accountItem'>
            <Link to="/account">Account</Link>
          </Menu.Item>
        </Menu>
        <Switch>
          <Route path="/about">
            <Suspense fallback={
                <div className="content">
                  <Space className="space_main">
                      <Loader /> 
                  </Space>
                </div>} >
                <About />
            </Suspense>          
          </Route>
          <Route path="/trips">
            <Suspense fallback={
                <div className="content">
                  <Space className="space_main">
                      <Loader /> 
                  </Space>
                </div>} >
                <ListTrips />
            </Suspense>
          </Route>
          <Route path="/airports">
            <Suspense fallback={
                <div className="content">
                  <Space className="space_main">
                      <Loader /> 
                  </Space>
                </div>} >
                <ListAirports />
            </Suspense>          
          </Route>
          <Route path="/aircraft">
            <Suspense fallback={
                <div className="content">
                  <Space className="space_main">
                      <Loader /> 
                  </Space>
                </div>} >
                <ListAircraft />
            </Suspense>              
          </Route>
          <Route  path="/login">
            <div className="content">
              <Login />
            </div>
          </Route>
          <Route path="/">
            <Suspense fallback={
                <div className="content">
                  <Space className="space_main">
                      <Loader /> 
                  </Space>
                </div>} >
              <Home />
            </Suspense>
          </Route>          
        </Switch>
      </Router>
    );
  }
}