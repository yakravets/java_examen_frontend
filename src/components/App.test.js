import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";

import App from './app';

export default class AppConsumer{
    constructor() {
        this.app = new App();
    }
}

it("renders correctly", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
});

beforeEach(() => {
        // Clear all instances and calls to constructor and all methods:
        App.mockClear();
    }
);

it('We can check if the consumer called the class constructor', () => {
    const appConsumer = new AppConsumer();
    expect(App).toHaveBeenCalledTimes(1);
});

const mockHandleClick = jest.fn();
jest.mock('./app', ()=>{
    return jest.fn().mockImplementation(() => {
        return {handleClick: mockHandleClick};
      });
}); 