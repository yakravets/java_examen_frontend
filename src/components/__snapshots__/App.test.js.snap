// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Menu
    mode="horizontal"
    onClick={[Function]}
    selectedKeys={
      Array [
        "home",
      ]
    }
  >
    <MenuItem
      icon={<GrMenu />}
      key="home"
    >
      <Link
        to="/"
      >
        Home
      </Link>
    </MenuItem>
    <MenuItem
      icon={<FaRegListAlt />}
      key="trips"
    >
      <Link
        to="/trips"
      >
        Trips
      </Link>
    </MenuItem>
    <MenuItem
      icon={<MdAirlineSeatReclineExtra />}
      key="airports"
    >
      <Link
        to="/airports"
      >
        Airports
      </Link>
    </MenuItem>
    <MenuItem
      icon={<GiCommercialAirplane />}
      key="aircraft"
    >
      <Link
        to="/aircraft"
      >
        Aircraft
      </Link>
    </MenuItem>
    <MenuItem
      icon={<AiOutlineInfoCircle />}
      key="about"
    >
      <Link
        to="/about"
      >
        About
      </Link>
    </MenuItem>
  </Menu>
  <Switch>
    <Route
      path="/about"
    >
      <Suspense
        fallback={
          <div
            className="content"
          >
            <Space
              className="space_main"
            >
              <Loader />
            </Space>
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Route>
    <Route
      path="/trips"
    >
      <Suspense
        fallback={
          <div
            className="content"
          >
            <Space
              className="space_main"
            >
              <Loader />
            </Space>
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Route>
    <Route
      path="/airports"
    >
      <Suspense
        fallback={
          <div
            className="content"
          >
            <Space
              className="space_main"
            >
              <Loader />
            </Space>
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Route>
    <Route
      path="/aircraft"
    >
      <Suspense
        fallback={
          <div
            className="content"
          >
            <Space
              className="space_main"
            >
              <Loader />
            </Space>
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Route>
    <Route
      path="/login"
    >
      <div
        className="content"
      >
        <Login />
      </div>
    </Route>
    <Route
      path="/"
    >
      <Suspense
        fallback={
          <div
            className="content"
          >
            <Space
              className="space_main"
            >
              <Loader />
            </Space>
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Route>
  </Switch>
</BrowserRouter>
`;
